src/ZipkinBundle/SpanNaming/Route/Naming.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace ZipkinBundle\SpanNaming\Route;
+
+use Symfony\Component\HttpFoundation\Request;
+use ZipkinBundle\SpanNaming\SpanNamingInterface;
+
+final class Naming implements SpanNamingInterface
+{
+    /**
+     * @var array
+     */
+    private $routes;
+
+    private function __construct(array $routes = [])
+    {
+        $this->routes = $routes;
+    }
+
+    public static function create($cacheDir)
+    {
+        $routes = @include CacheWarmer::buildOutputFilename($cacheDir);
+        return new self($routes === false ? [] : $routes);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getName(Request $request)
+    {
+        $routeName = $request->request->get('_route');
+
+        if (array_key_exists($routeName, $this->routes)) {
```

Comment:
fyi in brave we default to "get not_found" or "get redirected" when we are in a framework that supports routes, but there was no route found.

Reply:
This is not the same case. If `$routeName` is null then we return a `not_found`. If `$routeName != null` and it is not in the `routes` then there is something wrong and we fail gracefully.